### PR TITLE
solution for array method splice by @laverka

### PR DESCRIPTION
**Pull request by valeriitovstyk**

### CODE REVIEW

File: src/arrayMethodSplice.js [hunk 1]
```diff
@@ -5,7 +5,56 @@
...
+      removeFromBuffer();
+      putTogether(this);
+    }
+    if (argLength > 0) {
+      removeFromBuffer();
+      insertArgs(this);
+      putTogether(this);
+    }
+
+    function removeFromBuffer() {
```
Comment (YuriiHoliuk) [inline src/arrayMethodSplice.js]: if you need helper functions create pure functions without side effects outside a main function. Do not declare them inside applyCustomSplice

File: src/arrayMethodSplice.js [hunk 2]
```diff
@@ -5,7 +5,56 @@
  */
 function applyCustomSplice() {
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    const objectLength = this.length;
+    const argLength = items.length;
+    let removedItems = [];
+    let temp = [];
+
+    if (start < 0) start += objectLength;
```
[the remark this answers is not in the document]
Comment (valeriitovstyk) [reply src/arrayMethodSplice.js]: done

File: src/arrayMethodSplice.js [hunk 3]
```diff
@@ -5,7 +5,56 @@
...
+    if ((start * -1) > objectLength) {
```
[the remark this answers is not in the document]
Comment (valeriitovstyk) [reply src/arrayMethodSplice.js]: removed

File: src/arrayMethodSplice.js [hunk 4]
```diff
@@ -5,7 +5,56 @@
  */
 function applyCustomSplice() {
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    const objectLength = this.length;
```
[the remark this answers is not in the document]
Comment (valeriitovstyk) [reply src/arrayMethodSplice.js]: done

File: src/arrayMethodSplice.js [hunk 5]
```diff
@@ -5,7 +5,56 @@
...
+
+    if (start < 0) start += objectLength;
+    if ((start * -1) > objectLength) {
+      removedItems = [...this];
+      this.length = 0;
+      return removedItems;
+    }
+    if (start === 0) return this;
+
+    if (deleteCount > objectLength) deleteCount = 0;
```
[the remark this answers is not in the document]
Comment (valeriitovstyk) [reply src/arrayMethodSplice.js]: changed

File: src/arrayMethodSplice.js [hunk 6]
```diff
@@ -5,7 +5,56 @@
...
+    if (deleteCount === 0 && (!items)) return removedItems;
```
[the remark this answers is not in the document]
Comment (valeriitovstyk) [reply src/arrayMethodSplice.js]: removed

File: src/arrayMethodSplice.js [hunk 7]
```diff
@@ -5,7 +5,56 @@
...
+    function removeFromBuffer() {
+      temp = removedItems.slice();
+      removedItems = [];
+      for (let i = 0; i < deleteCount; i++) {
+        removedItems.push(temp.pop());
+      }
+      removedItems.reverse();
+    }
+
+    function insertArgs(obj) {
```
[the remark this answers is not in the document]
Comment (valeriitovstyk) [reply src/arrayMethodSplice.js]: got rid of functions

File: src/arrayMethodSplice.js [hunk 8]
```diff
@@ -5,7 +5,56 @@
...
+      items.forEach(item => obj.push(item));
+    }
+
+    function putTogether(obj) {
```
[the remark this answers is not in the document]
Comment (valeriitovstyk) [reply src/arrayMethodSplice.js]: there is no functions inside

File: src/arrayMethodSplice.js [hunk 9]
```diff
@@ -5,7 +5,56 @@
...
+      let tempLength = temp.length;
```
[the remark this answers is not in the document]
Comment (valeriitovstyk) [reply src/arrayMethodSplice.js]: done

File: src/arrayMethodSplice.js [hunk 10]
```diff
@@ -5,7 +5,56 @@
...
+      for (let j = 0; j < tempLength; j++) {
```
[the remark this answers is not in the document]
Comment (valeriitovstyk) [reply src/arrayMethodSplice.js]: done

File: src/arrayMethodSplice.js (hunk 1 above)
Comment (valeriitovstyk) [reply src/arrayMethodSplice.js]: there is no functions inside

File: src/arrayMethodSplice.js [hunk 11]
```diff
@@ -5,7 +5,41 @@
...
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    const initialLength = this.length;
+    const argLength = items.length;
+    let removedItems = [];
+    let buffer = [];
+    let actualStartPosition = start;
+    let actualDeleteCount = deleteCount;
+
+    if (actualStartPosition === 0) return this;
```
Comment (YuriiHoliuk) [inline src/arrayMethodSplice.js]: you always should return a new array
also, what if you have items

File: src/arrayMethodSplice.js [hunk 12]
```diff
@@ -5,7 +5,41 @@
...
+    if (actualStartPosition < 0) actualStartPosition += initialLength;
+    if ((!arguments.length) ||
+      (actualDeleteCount === 0 && (!items)) ||
+      (actualStartPosition > initialLength)) {
```
Comment (YuriiHoliuk) [inline src/arrayMethodSplice.js]: unneeded parenthesis, format conditions

File: src/arrayMethodSplice.js [hunk 13]
```diff
@@ -5,7 +5,41 @@
...
+    const initialLength = this.length;
+    const argLength = items.length;
+    let removedItems = [];
+    let buffer = [];
+    let actualStartPosition = start;
+    let actualDeleteCount = deleteCount;
+
+    if (actualStartPosition === 0) return this;
+    if (actualStartPosition < 0) actualStartPosition += initialLength;
+    if ((!arguments.length) ||
```
Comment (YuriiHoliuk) [inline src/arrayMethodSplice.js]: Do you mean argLength?
do not use `arguments` variable

File: src/arrayMethodSplice.js [hunk 14]
```diff
@@ -5,7 +5,41 @@
...
+      (actualDeleteCount === 0 && (!items)) ||
+      (actualStartPosition > initialLength)) {
+      return removedItems;
+    }
+
+    if ((actualStartPosition * -1) > initialLength) {
```
Comment (YuriiHoliuk) [inline src/arrayMethodSplice.js]: unneeded parenthesis

**Pull request merged**